### PR TITLE
Datasync next

### PR DESCRIPTION
**Pull request by wtrocki**

### CODE REVIEW

**Comment by wtrocki** [comment]
@machi1990 This is still work in progress but we should use this branch for the collaboration on phase 1. No force pushes :D

**Comment by machi1990** [comment]
> @machi1990 This is still work in progress but we should use this branch for the collaboration on phase 1. No force pushes :D

May the force be with you :dagger: :-D

**Comment by machi1990** [comment]
@wtrocki I was planning to update the base image to `registry.redhat.io/rhel8/nodejs-12` but I am getting authorization issues when trying to build the new Datasync image.

```bash
docker build . -t test
Sending build context to Docker daemon  125.4kB
Step 1/7 : FROM registry.redhat.io/rhel8/nodejs-12
Get https://registry.redhat.io/v2/rhel8/nodejs-12/manifests/latest: unauthorized: Please login to the Red Hat Registry using your Customer Portal credentials. Further instructions can be found here: https://access.redhat.com/RegistryAuthentication
```

I'll probably need to sign in to the `registry.redhat.io` register for me to complete my testing, but wanted to know if we've access to the credentials?

**Comment by wtrocki** [comment]
> I'll probably need to sign in to the registry.redhat.io register for me to complete my testing, but wanted to know if we've access to the credentials?

Screenshot 2020-09-18 at 12 46 09

**Comment by wtrocki** [comment]
Ah.. Those images are not comming from quay
they coming from different registry. I would use some other images if possible:
https://catalog.redhat.com/software/containers/rhscl/nodejs-10-rhel7/5baa36dbbed8bd6ee81a09c7

I know some of them are available on vanilla openshift so they should work without dev account. 
Logging to dev account is ok, but it will be too much hassle for CI etc.

**Comment by machi1990** [comment]
> Ah.. Those images are not comming from quay
> they coming from different registry. I would use some other images if possible:

Nope, they are coming from redhat.registry.io and the registry needs one to be logged in.

> https://catalog.redhat.com/software/containers/rhscl/nodejs-10-rhel7/5baa36dbbed8bd6ee81a09c7
> 
> I know some of them are available on vanilla openshift so they should work without dev account.

I could update the base image to node:14 and revisit this part later on as I am currently checking on the openshift template for mongo. 

> Logging to dev account is ok, but it will be too much hassle for CI etc.

Exactly.

**Comment by machi1990** [comment]
@wtrocki I am looking to test https://github.com/aerogear/datasync-starter/pull/444/commits/5758ab0ece6d7c343bd144df2facb6835cd4c7a5 next. But if you can review it, would be very much appreciated.

**Comment by wtrocki** [comment]
@machi1990 This docker image will not work with s2i process for building container. 
We might avoid using S2i for the moment and work on the template only, but generally this will be not red hat way of doing things. Good for demo but it might prevent us from merging this change.

**Comment by machi1990** [comment]
> @machi1990 This docker image will not work with s2i process for building container.
> We might avoid using S2i for the moment and work on the template only, but generally this will be not red hat way of doing things. Good for demo but it might prevent us from merging this change.

Thanks for checking it out. Sure, we can ignore s2i for the moment unless otherwise we'll want to bring it into the demo (which I think we should not). 

Just for my own knowledge: I've never used s2i before so I might not have an idea which part of the Dockerfile may not work with s2i. Is it the multistage part?

**Comment by wtrocki** [comment]
> Just for my own knowledge: I've never used s2i before so I might not have an idea which part of the Dockerfile may not work with s2i. Is it the multistage part?

@machi1990 This is not technical. We will use this image. 

We work in demo setup so I guess we will not merge this branch for a while

**Comment by machi1990** [comment]
> > Just for my own knowledge: I've never used s2i before so I might not have an idea which part of the Dockerfile may not work with s2i. Is it the multistage part?
> 
> @machi1990 This is not technical. 

Thanks. I got you.

File: .openshift/datasync-app-template.yml [hunk 1]
```diff
@@ -78,8 +134,13 @@ objects:
           - containerPort: 4000
             protocol: TCP
           resources: {}
+          volumeMounts:
+            - name: config
+              mountPath: "/usr/src/model"
```
Comment (machi1990) [inline .openshift/datasync-app-template.yml:108]: This path is based on https://github.com/aerogear/datasync-starter/pull/444/commits/550f98ace06e5b0c059dc71cce9e41ed213b4564#diff-9b7885c8f82945595429185389c21936R19

**Comment by wtrocki** [comment]
@machi1990 I think we should theat this branch as feature branch with no direct commits from now on since it will be hard to review it and track progress.

**Comment by machi1990** [comment]
> @machi1990 I think we should theat this branch as feature branch with no direct commits from now on since it will be hard to review it and track progress.

Sure. I've a small change that make the model in a config map a variable. I'll push it in a separate branch then

**Pull request closed without merging**